openpgp.d.ts

```diff
@@ -337,13 +338,20 @@ interface Config {
   rejectMessageHashAlgorithms: Set<enums.hash>;
   rejectPublicKeyAlgorithms: Set<enums.publicKey>;
   rejectCurves: Set<enums.curve>;
+  hardwareKeys: HardwareKeys | null;
 }
 export var config: Config;
 
 // PartialConfig has the same properties as Config, but declared as optional.
 // This interface is relevant for top-level functions, which accept a subset of configuration options
 interface PartialConfig extends Partial<Config> {}
 
+declare abstract class HardwareKeys {
```

Comment:
As Daniel [mentioned](https://github.com/openpgpjs/openpgpjs/pull/1567#discussion_r1084389921), I think this should be an interface, as it's more generic. We are using abstract classes if we are extending them as part of the definitions or if we need other OOP features (e.g. `instanceof HardwareKeys`), which doesn't seem to be the case.

Reply:
I agree. There is no blanket implementation either. I remember I wanted to model it, so that the JavaScript's abstract class would have the same interface to keep it somewhat compatible between languages. I see the point in having this Typescript only, as a proper type checker.
I will change that abstract class to interface as proposed.

Reply:
Changed as suggested to interface. Tests are passing:
- browser
- node (without Webcrypt)
- `npm run test-type-definitions`